moveit_ros/planning/planning_request_adapter_plugins/src/chomp_optimizer_adapter.cpp

```diff
@@ -0,0 +1,242 @@
+    }
+    if (!nh_.getParam("joint_update_limit", params_.joint_update_limit_))
+    {
+      params_.joint_update_limit_ = 0.1;
+      ROS_INFO_STREAM("Param joint_update_limit was not set. Using default value: " << params_.joint_update_limit_);
+    }
+    if (!nh_.getParam("min_clearence", params_.min_clearence_))
+    {
+      params_.min_clearence_ = 0.2;
+      ROS_INFO_STREAM("Param min_clearence was not set. Using default value: " << params_.min_clearence_);
+    }
+    if (!nh_.getParam("collision_threshold", params_.collision_threshold_))
+    {
+      params_.collision_threshold_ = 0.07;
+      ROS_INFO_STREAM("Param collision_threshold_ was not set. Using default value: " << params_.collision_threshold_);
+    }
+    if (!nh_.getParam("use_stochastic_descent", params_.use_stochastic_descent_))
+    {
+      params_.use_stochastic_descent_ = true;
+      ROS_INFO_STREAM(
+          "Param use_stochastic_descent was not set. Using default value: " << params_.use_stochastic_descent_);
+    }
+    if (!nh_.getParam("trajectory_initialization_method", params_.use_stochastic_descent_))
+    {
+      params_.trajectory_initialization_method_ = std::string("OMPL");
+      ROS_INFO_STREAM("Param trajectory_initialization_method was not set. Using New value as: "
+                      << params_.trajectory_initialization_method_);
+    }
+  }
+
+  virtual std::string getDescription() const
+  {
+    return "CHOMP Optimizer !!!!";
+  }
+
+  virtual bool adaptAndPlan(const PlannerFn& planner, const planning_scene::PlanningSceneConstPtr& planning_scene,
+                            const planning_interface::MotionPlanRequest& req,
+                            planning_interface::MotionPlanResponse& res,
+                            std::vector<std::size_t>& added_path_index) const
+  {
+    // following call to planner() calls the OMPL planner and stores the trajectory inside the MotionPlanResponse res
+    // variable which is then used by CHOMP for optimization of the computed trajectory
+    bool solved = planner(planning_scene, req, res);
+
+    // create a hybrid collision detector to set the collision checker as hybrid
+    collision_detection::CollisionDetectorAllocatorPtr hybrid_cd(
+        collision_detection::CollisionDetectorAllocatorHybrid::create());
+
+    ROS_INFO_STREAM("Configuring Planning Scene for CHOMP ....");
+    planning_scene->setActiveCollisionDetector_ConstVersion(hybrid_cd, true);
+
+    chomp::ChompPlanner chompPlanner;
+    planning_interface::MotionPlanDetailedResponse res_detailed;
+    moveit_msgs::MotionPlanDetailedResponse res2;
+    moveit_msgs::MotionPlanRequest req_moveit_msgs;
+
+    // populate the trajectory to pass to CHOMPPlanner::solve() method. Obtain trajectory from OMPL's
+    // planning_interface::MotionPlanResponse object and put / populate it in the
+    // moveit_msgs::MotionPlanDetailedResponse object
+    moveit_msgs::RobotTrajectory trajectory_msgs2;
```

Comment:
better variable name than "2"

Reply:
changed to `trajectory_msg_from_response`